articles/architecture-scenarios/b2c/user-provisioning.md

```diff
@@ -0,0 +1,94 @@
+
+### Use your own identity data store
+
+If you want to use your own identity store, perhaps because you’ve got applications which you aren’t ready to migrate or can’t be migrated, use Auth0’s [Database Connections](/connections/database/custom-db) to make a connection to an existing identity store. 
+
+### Use Auth0's identity data store
+
+Auth0 provides identity data storage that you can use to store and manage your users' credentials. You can also migrate legacy data to Auth0 using Auth0's user migration tools. 
+
+## User migration
+
+If you decide to use Auth0, you can [migrate users](/users/concepts/overview-user-migration) in one of two ways: 
+
+* With bulk migration to bring all your data at once
+* With progressive automatic migration to bring your data over time
+
+Keep in mind that there are multiple approaches to user migration - you are not obligated to choose only one option. For example, you might opt for a two-phased approach to migration, beginning with automatic migration and following it up with bulk migration for the remaining users. See [User Migration Scenarios](/users/references/user-migration-scenarios) for additional options.
+
+### Bulk migration
+
+There are two tools you can use when migrating users in bulk:
+
+* The Management API
+* The User Import/Export Extension
+
+In most cases, we recommend that you use the [Management API](/users/concepts/overview-user-migration#bulk-user-imports-with-the-management-api), since this option provides you with more flexibility and control. We only recommend the [User Import/Export extension](https://auth0.com/docs/users/concepts/overview-user-migration#migrate-users-with-the-user-import-export-extension) in simple cases, such as the movement of a few users.
+
+Please note that your users will need to reset their passwords once you've migrated their accounts to Auth0 using one of the bulk migration processes.
+
+### Automatic migration
+
+[Automatic migration](/users/guides/configure-automatic-migration) allows users to be migrated incrementally. Users whose accounts are migrated to Auth0 using this method will not need to reset their passwords afterward. 
+
+## User sign up
+
+Auth0 uses database connections to store the user ID, password, and username collected from new users during the sign-up process. You can configure the database connection policies governing things such as minimum username length or password strength and complexity using the Auth0 dashboard.  
+
+Auth0 offers [Lock](/libraries), which providers a user interface that includes signup functionality out of the box. Lock supports a some customization to its appearance, but if you would like to make extensive changes to the login widget to achieve your branding goals, we recommend that you use [Universal Login](/universal-login) instead. For a comparison of the capabilities, see [Lock vs. a Custom UI](/libraries/when-to-use-lock).
+
```

Comment:
>Auth0 offers Lock...

This comparison isn't really right. If you want to make extensive changes, you would go from Lock to Auth0.js.

Universal Login isn't the custom alternative to Lock.

However, we really should be proffering Universal Login first, and noting that if it doesn't meet your needs, we also have libraries that can be embedded in your application, such as Lock (premade login widget) and Auth0.js (a library with which you can build your own login UI).

Reply:
Good. That confirms what I thought.